rust/cardano-blockchain-types/src/metadata/cip36/mod.rs

```diff
@@ -45,12 +105,21 @@ impl Cip36 {
     /// decoded.
     pub fn new(
         block: &MultiEraBlock, txn_idx: TxnIndex, is_catalyst_strict: bool,
-    ) -> anyhow::Result<Self> {
+    ) -> Result<Cip36, Cip36Error> {
```

Comment:
I'm thinking about whether we should distinguish between three different situations:
- There is no `Cip36` in the transaction. Technically it isn't an error, so it can be represented as `Ok(None)`.
- The data is completely corrupt: `Err(...)`.
- Cip36 is present and was at least partially decoded: `Ok(Cip36)`. In that case the problem report inside will indicate if the decoding was completely successful.

Will it just unnecessarily complicate the handling on the call site?..

Reply:
Agree, should return None if there is no CIP36 in the given index